Installation issues with sub command strip and git submodule update

Some issues came up during the installation. Check below the error:

```
 cmake -DSKIP_OPENMP=YES -DSKIP_MPI=YES ..
-- The Fortran compiler identification is GNU 4.8.5
-- The C compiler identification is GNU 4.8.5
-- Check for working Fortran compiler: /usr/bin/f95
-- Check for working Fortran compiler: /usr/bin/f95  -- works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether /usr/bin/f95 supports Fortran 90
-- Checking whether /usr/bin/f95 supports Fortran 90 -- yes
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- *** Setting default install prefix to /home/users/ajornet/pFUnit/build/installed.
-- *** Override with -DCMAKE_INSTALL_PREFIX=<path>.
-- Performing Test _LOGICAL_DEFAULT_KIND: SUCCESS (value=4)
-- Performing Test _INT_DEFAULT_KIND: SUCCESS (value=4)
-- Performing Test _ISO_INT8: SUCCESS (value=1)
-- Performing Test _ISO_INT16: SUCCESS (value=2)
-- Performing Test _ISO_INT32: SUCCESS (value=4)
-- Performing Test _ISO_INT64: SUCCESS (value=8)
-- Performing Test _REAL_DEFAULT_KIND: SUCCESS (value=4)
-- Performing Test _DOUBLE_DEFAULT_KIND: SUCCESS (value=8)
-- Performing Test _ISO_REAL32: SUCCESS (value=4)
CMake Error at cmake/CheckFortranSource.cmake:13 (string):
  string sub-command STRIP requires two arguments.
Call Stack (most recent call first):
  cmake/CheckCompilerCapabilities.cmake:57 (CHECK_FORTRAN_SOURCE_RUN)
  CMakeLists.txt:55 (include)

 -- Performing Test _REAL32_IEEE_SUPPORT: SUCCESS (value=)
-- Performing Test _ISO_REAL64: SUCCESS (value=8)
CMake Error at cmake/CheckFortranSource.cmake:13 (string):
  string sub-command STRIP requires two arguments.
Call Stack (most recent call first):
  cmake/CheckCompilerCapabilities.cmake:57 (CHECK_FORTRAN_SOURCE_RUN)
  CMakeLists.txt:55 (include)


 -- Performing Test _REAL64_IEEE_SUPPORT: SUCCESS (value=)
-- Performing Test _ISO_REAL128: SUCCESS (value=10)
CMake Error at cmake/CheckFortranSource.cmake:13 (string):
  string sub-command STRIP requires two arguments.
Call Stack (most recent call first):
  cmake/CheckCompilerCapabilities.cmake:57 (CHECK_FORTRAN_SOURCE_RUN)
  CMakeLists.txt:55 (include)


-- Performing Test _REAL128_IEEE_SUPPORT: SUCCESS (value=)
You need to run this command from the toplevel of the working tree.
CMake Error at cmake/build_submodule.cmake:26 (message):
  git submodule update --init failed with 1, please checkout submodules
Call Stack (most recent call first):
  extern/CMakeLists.txt:2 (build_submodule)


-- Configuring incomplete, errors occurred!
See also "..../pFUnit/build/CMakeFiles/CMakeOutput.log".
```

If works after applying the following fixes:
```
git diff origin/master
diff --git a/cmake/CheckFortranSource.cmake b/cmake/CheckFortranSource.cmake
index 0c0883e..7259c29 100644
--- a/cmake/CheckFortranSource.cmake
+++ b/cmake/CheckFortranSource.cmake
@@ -10,7 +10,7 @@ macro (CHECK_FORTRAN_SOURCE_RUN file var)
 
   # Successful runs return "0", which is opposite of CMake sense of "if":
   if (NOT run)
-    string(STRIP ${${var}} ${var})
+    string(STRIP "${${var}}" ${var})
     if (NOT CMAKE_REQUIRED_QUIET)
       message(STATUS "Performing Test ${var}: SUCCESS (value=${${var}})")
     endif ()
diff --git a/cmake/build_submodule.cmake b/cmake/build_submodule.cmake
index 4b55ce9..4f70542 100644
--- a/cmake/build_submodule.cmake
+++ b/cmake/build_submodule.cmake
@@ -19,7 +19,7 @@ macro (build_submodule submodule)
        if (GIT_SUBMODULE)
          execute_process (
            COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive
-           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
            RESULT_VARIABLE GIT_SUBMODULE_RESULT
            )
          if (NOT GIT_SUBMODULE_RESULT EQUAL "0")
```

I think so because after defining:
```
export FC=ifort
```

I got the following output:

```
[ajornet@obelix2 build]$ cmake -DSKIP_OPENMP=YES -DSKIP_MPI=YES ..
-- The Fortran compiler identification is Intel 18.0.1.20171018
-- The C compiler identification is GNU 4.8.5
...

```
So the Intel fortran compiler is detected. 

After running the command again with no modification done and the key above defined:
```
[ajornet@obelix2 build]$ cmake -DSKIP_OPENMP=YES -DSKIP_MPI=YES ..
-- The Fortran compiler identification is Intel 18.0.1.20171018
-- The C compiler identification is GNU 4.8.5
-- Check for working Fortran compiler: /home/share/unix_files/intel-fcomp-2018/compilers_and_libraries_2018.1.163/linux/bin/intel64/ifort
-- Check for working Fortran compiler: /home/share/unix_files/intel-fcomp-2018/compilers_and_libraries_2018.1.163/linux/bin/intel64/ifort  -- works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether /home/share/unix_files/intel-fcomp-2018/compilers_and_libraries_2018.1.163/linux/bin/intel64/ifort supports Fortran 90
-- Checking whether /home/share/unix_files/intel-fcomp-2018/compilers_and_libraries_2018.1.163/linux/bin/intel64/ifort supports Fortran 90 -- yes
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- *** Setting default install prefix to /home/users/ajornet/pFUnit/build/installed.
-- *** Override with -DCMAKE_INSTALL_PREFIX=<path>.
-- Performing Test _LOGICAL_DEFAULT_KIND: SUCCESS (value=4)
-- Performing Test _INT_DEFAULT_KIND: SUCCESS (value=4)
-- Performing Test _ISO_INT8: SUCCESS (value=1)
-- Performing Test _ISO_INT16: SUCCESS (value=2)
-- Performing Test _ISO_INT32: SUCCESS (value=4)
-- Performing Test _ISO_INT64: SUCCESS (value=8)
-- Performing Test _REAL_DEFAULT_KIND: SUCCESS (value=4)
-- Performing Test _DOUBLE_DEFAULT_KIND: SUCCESS (value=8)
-- Performing Test _ISO_REAL32: SUCCESS (value=4)
-- Performing Test _REAL32_IEEE_SUPPORT: SUCCESS (value=T)
-- Performing Test _ISO_REAL64: SUCCESS (value=8)
-- Performing Test _REAL64_IEEE_SUPPORT: SUCCESS (value=T)
-- Performing Test _ISO_REAL128: SUCCESS (value=16)
-- Performing Test _REAL128_IEEE_SUPPORT: SUCCESS (value=T)
You need to run this command from the toplevel of the working tree.
CMake Error at cmake/build_submodule.cmake:26 (message):
  git submodule update --init failed with 1, please checkout submodules
Call Stack (most recent call first):
  extern/CMakeLists.txt:2 (build_submodule)


-- Configuring incomplete, errors occurred!
See also "..../pFUnit/build/CMakeFiles/CMakeOutput.log".
```

--- a/cmake/CheckFortranSource.cmake
+++ b/cmake/CheckFortranSource.cmake
@@ -10,7 +10,7 @@ macro (CHECK_FORTRAN_SOURCE_RUN file var)
 
   # Successful runs return "0", which is opposite of CMake sense of "if":
   if (NOT run)
-    string(STRIP ${${var}} ${var})
+    string(STRIP "${${var}}" ${var})
     if (NOT CMAKE_REQUIRED_QUIET)
       message(STATUS "Performing Test ${var}: SUCCESS (value=${${var}})")
     endif ()

